### ISSUE TITLE
PCA for new/test set

### ISSUE THREAD

**Issue by svkucheryavski** [yzontov/pls-da#49]
When new/test set is used it can be a good idea to:

- instead of building a separate PCA model apply one created for calibration set. If user did not create any, do it first and then apply. Or do not show PCA tab for new/test data if no PCA model was crated for calibration set
- show calibration data in PCA results as gray points so we can see the relationship between the two sets

**Comment by yzontov** [comment]
Идея в том, что PCA на вкладке Data - это, как и все на этой вкладке, только средство анализа исходных данных. Регрессию на ГК программа не делает, поэтому на этом этапе ее работы нет никакой связи между объектами DataSet, и нельзя сказать, что такой-то набор является калибровочным для другого.

**Comment by svkucheryavski** [comment]
Да, но идея в том, что тестовый и новый наборы данных должны как-то соотносится с калибровочным. И МГК сделанная на калиброчовном наборе позволила бы это показать. Немного странно на мой взгляд делать независимую МГК модель для проверочного набора. Впрочем, я не настаиваю, это мое мнение так сказать.

**Comment by yzontov** [comment]
Как бы Вы посоветовали лучше связать наборы между собой? Просто сейчас пользователь может сделать их несколько, как калибровочных, так и проверочных. В принципе признак "проверочный" сейчас - это в основном просто метка, на вкладке Predict доступны все наборы с данными подходящей размерности. 
Т.е. сейчас единственная связь между наборами - это размерность данных и метки Calibration/Test, а в случае нескольких калибровочных наборов этого будет не достаточно. 
Дать пользователю возможность указать связанный калибровочный набор при создании тестового?

**Comment by svkucheryavski** [comment]
Я бы сделал так. 

Первый вариант запретить использовать больше чем один калибровочный набор в программе. Его должен выбрать пользователь в самом начале (первый выбранный датасет). И не давать возможности его менять. Хочешь сделать модель с новым калибровочным набором - перезапускай программу. Либо нажимай кнопку ресет чтобы все результаты стерлись (сначала показать диалог с возможностью их сохранения) и снова появилась возможность выбрать калибровочный датасет.

Второй вариант сделать поддержку нескольких калибровочных наборов но в этом случае все модели и тестовые наборы должны быть привязаны к калибровочному. По сути калибровочный набор образует кейс - вершину дерева - к которому присоединяются все остальные объекты в виде новых и тестовых наборов, модели, препроцессинга, результатов и т.п.

На мой взгляд первый вариант гораздо проще и покрывает 90% надобностей. Посоветуйтесь еще с ОЕ и АЛ.
